### PR TITLE
Fix wrong contract account id for `instantiate`

### PR DESCRIPTION
**Pull request by lean-apple**
Closes https://github.com/paritytech/cargo-contract/issues/777. 

The last contract account id was used instead of the first one for the Instantiated Event. 

A `find_last` method was added in the events_types of [subxt](https://github.com/paritytech/subxt) to lighten the current code and for block_types, [here](https://github.com/paritytech/subxt/pull/825).

### CODE REVIEW

File: crates/cargo-contract/src/cmd/extrinsics/instantiate.rs [hunk 1]
```diff
@@ -266,13 +266,33 @@ impl Exec {
             .find_first::<api::contracts::events::CodeStored>()?
             .map(|code_stored| code_stored.code_hash);
 
-        let instantiated = result
-            .find_first::<api::contracts::events::Instantiated>()?
-            .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
+        // //TODO:Remove the useless part
+        // let instantiated = result
+        //     .find_first::<api::contracts::events::Instantiated>()?
+        //     .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
```
Comment (ascjones) [inline crates/cargo-contract/src/cmd/extrinsics/instantiate.rs:272]:
```suggestion
```
Remove commented out code

File: crates/cargo-contract/src/cmd/extrinsics/instantiate.rs [hunk 2]
```diff
@@ -266,13 +266,33 @@ impl Exec {
...
+            .filter_map(|ev| {
+                ev.and_then(|ev| {
+                    ev.as_event::<api::contracts::events::Instantiated>()
+                        .map_err(Into::into)
+                })
+                .transpose()
+            })
+            .last()
+            .transpose()?
+            .ok_or_else(|| anyhow!("Failed to find Last Instantiated event"))?;
```
Comment (ascjones) [inline crates/cargo-contract/src/cmd/extrinsics/instantiate.rs:284]:
```suggestion
            .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
```
I think this case is it failed to find *any* instantiated event

File: crates/cargo-contract/src/cmd/extrinsics/instantiate.rs [hunk 3]
```diff
@@ -266,13 +266,33 @@ impl Exec {
...
+
+        // println!("Test Code Hash Last Instantiated'{:?}'", last_instantiated.contract.to_string());
```
Comment (ascjones) [inline crates/cargo-contract/src/cmd/extrinsics/instantiate.rs:286]:
```suggestion
```
Remove commented out code

**Review by ascjones — APPROVED** [review]
Assuming it works okay by passing the repro in https://github.com/paritytech/cargo-contract/issues/777. LGTM!

**Comment by cmichi** [comment]
Oha, I somehow thought we had merged a hotfix for this issue.

@ascjones We should issue a patch release with this one and the template fix.

@lean-apple Is the PR still WIP? If not change the title please. Please also update the first sentence in the description to "Closes $link_to_issue.". When we merged the PR GitHub will then automatically close the issue.

**Review by cmichi — APPROVED** [review]
Thank you, Lea!

**Comment by cmichi** [comment]
@lean-apple Have you tried to replicate the issue from https://github.com/paritytech/cargo-contract/issues/777 again? Can it no longer be replicated with your PR?

**Pull request merged**